File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty())
            return 0;

        int least_price = prices[0];
        int ans = 0;

        for (int cur_price : prices) {

            int ret = cur_price - least_price;
            ans = max(ans, ret);

            least_price = min(cur_price, least_price);
        }

        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let min = Number.MAX_SAFE_INTEGER;
    let ans = 0 ;
    for(let i of prices) {
        min = Math.min(min, i);
        ans = Math.max(ans, i - min);
        // console.log(ans)
    }
    return Math.max(ans , 0);
};